src/nodes.coffee

```diff
@@ -981,54 +981,37 @@ exports.Obj = class Obj extends Base
     if @generated
       for node in props when node instanceof Value
         node.error 'cannot have an implicit value in an implicit object'
-    break for prop, dynamicIndex in props when (prop.variable or prop).base instanceof Parens
-    hasDynamic  = dynamicIndex < props.length
-    idt         = o.indent += TAB
-    lastNoncom  = @lastNonComment @properties
+    idt        = o.indent += TAB
+    lastNoncom = @lastNonComment @properties
     answer = []
-    if hasDynamic
-      oref = o.scope.freeVariable 'obj'
-      answer.push @makeCode "(\n#{idt}#{oref} = "
-    answer.push @makeCode "{#{if props.length is 0 or dynamicIndex is 0 then '}' else '\n'}"
+    answer.push @makeCode "{#{if props.length is 0 then '}' else '\n'}"
     for prop, i in props
-      if i is dynamicIndex
-        answer.push @makeCode "\n#{idt}}" unless i is 0
-        answer.push @makeCode ',\n'
-      join = if i is props.length - 1 or i is dynamicIndex - 1
+      join = if i is props.length - 1
         ''
       else if prop is lastNoncom or prop instanceof Comment
         '\n'
       else
         ',\n'
       indent = if prop instanceof Comment then '' else idt
-      indent += TAB if hasDynamic and i < dynamicIndex
       if prop instanceof Assign
         if prop.context isnt 'object'
           prop.operatorToken.error "unexpected #{prop.operatorToken.value}"
         if prop.variable instanceof Value and prop.variable.hasProperties()
           prop.variable.error 'invalid object key'
       if prop instanceof Value and prop.this
         prop = new Assign prop.properties[0].name, prop, 'object'
-      if prop not instanceof Comment
-        if i < dynamicIndex
-          if prop not instanceof Assign
-            prop = new Assign prop, prop, 'object'
+      if prop not instanceof Comment and prop not instanceof Assign
+        if prop.isComplex()
```

Comment:
Previously, dynamic keys were being detected with...
```coffee
(prop.variable or prop).base instanceof Parens
```
... but this seems simpler. I don't know all the ins and outs though, so I might be missing something.